variables.tf

```
@@ -0,0 +1,224 @@
+variable "public_ip_config" {
+  type = object({
+    enabled           = bool
+    allocation_method = optional(string, "Static")
+  })
+  default = {
+    enabled = false
+  }
+  validation {
+    condition     = contains(["Static", "Dynamic"], var.public_ip_config.allocation_method)
+    error_message = "Allocation method must be Static or Dynamic"
+  }
+  description = <<-DOC
+  ```
+    enabled: Optionally select true if a public ip should be created. Defaults to false.
+    allocation_method: The allocation method of the public ip that will be created. Defaults to static.      
+  ```
+  DOC
+}
```

```suggestion
variable "public_ip_config" {
  type = object({
    enabled           = bool
    allocation_method = optional(string, "Static")
  })
  default = {
    enabled = false
  }
  validation {
    condition     = contains(["Static", "Dynamic"], var.public_ip_config.allocation_method)
    error_message = "Allocation method must be Static or Dynamic"
  }
  description = <<-DOC
  ```
    enabled: Optionally select true if a public ip should be created. Defaults to false.
    allocation_method: The allocation method of the public ip that will be created. Defaults to static.      
  ```
  DOC

nullable = false
}
```

---

The comment markdown is a bit confused by the markdown inside the code :)

Will add nullable = false